dev-libs/libprelude/libprelude-4.0.0.ebuild

```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+PYTHON_COMPAT=( python{2_7,3_{4,5,6}} )
+
+USE_RUBY="ruby21 ruby22 ruby23 ruby24"
+
+inherit autotools python-r1 ruby-single
+
+DESCRIPTION="Prelude-SIEM Framework Library"
+HOMEPAGE="https://www.prelude-siem.org"
+SRC_URI="https://www.prelude-siem.org/pkg/src/${PV}/${P}.tar.gz"
+
+LICENSE="GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="lua perl python ruby"
+
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+RDEPEND="net-libs/gnutls:=
+	lua? ( dev-lang/lua:* )
+	perl? ( dev-lang/perl:= virtual/perl-ExtUtils-MakeMaker )
+	python? ( ${PYTHON_DEPS} )
+	ruby? ( ${RUBY_DEPS} )"
+
+DEPEND="${RDEPEND}
+	sys-devel/flex
+	virtual/yacc
+	>=dev-lang/swig-3.0.7
+	virtual/pkgconfig"
+
+src_prepare() {
+	default
+
+	# Avoid null runpaths in Perl bindings.
+	sed -e 's/ LD_RUN_PATH=""//' -i "${S}/bindings/Makefile.am" || die "sed failed"
+
+	eautomake
+}
+
+src_configure() {
+	local python2_configure=--without-python2
+	local python3_configure=--without-python3
+
+	python_configure() {
+		if [[ ${EPYTHON} == python2* ]]; then
+			python2_configure=--with-python2
+		elif [[ ${EPYTHON} == python3* ]]; then
+			python3_configure=--with-python3
+		fi
+	}
+
+	if use python; then
+		python_foreach_impl python_configure
```

Comment:
You still don't have python prepared before the `econf` call. If you insist on doing it like this, you need to call `python_setup`.

Once you do that, I'm pretty sure configure will detect only Python 2 or Python 3, rendering the whole loop unnecessary. If it ignores the other Python missing, just pass `--with-python2 --with-python3` and let it figure out which one it is. If it dies, inline the check here to select 2 vs 3.